gort.go

```diff
@@ -76,6 +74,8 @@ func installAdapters() error {
 func startGort(ctx context.Context, configFile string, verboseCount int) error {
 	setLoggerVerbosity(verboseCount)
 
+	go catchSignals()
```

Comment:
What happens if SIGHUP is sent before initializeConfig returns?

Reply:
Good thinking! 

Both operations find their way to `config.Reload`, which locks on `configMutex`. However, the specifics largely depend on whether `signal.Notify` has completed by the time the `SIGHUP` signal comes in.

* If it HAS NOT, then the `SIGHUP` will be handled according to default behavior, which will cause Gort to exit. 
* If it HAS, then the signal will cause `config.Reload` to be called, which will block on the mutex resulting in two calls to `Reload`. If the underlying config file hasn't changed (per checksum) then the second call will be ignored.
 
In other words, nothing unusual. :)